### PR TITLE
[typescript/en-us] fix names to not collide and point out errors

### PR DESCRIPTION
**Pull request by vitiral**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

**Comment by vitiral** [comment]
Note: `name` is particularily problematic:

```
basics/yminutes.ts:190:5 - error TS2451: Cannot redeclare block-scoped variable 'name'.

190 let name = 'Tyrone';
        ~~~~

  ../../../../usr/local/lib/node_modules/typescript/lib/lib.dom.d.ts:19627:15
    19627 declare const name: never;
                        ~~~~
    'name' was also declared here.

../../../../usr/local/lib/node_modules/typescript/lib/lib.dom.d.ts:19627:15 - error TS2451: Cannot redeclare block-scoped variable 'name'.

19627 declare const name: never;
                    ~~~~

  basics/yminutes.ts:190:5
    190 let name = 'Tyrone';
            ~~~~
    'name' was also declared here.


Found 2 errors.
```

File: typescript.html.markdown [hunk 1]
```diff
@@ -54,6 +54,11 @@ function bigHorribleAlert(): void {
   alert("I'm a little annoying box!");
 }
 
+// Declaring a variable twice in the same scope results in an error.
+let myVar: int = 1;
+let myVar: int = 2; // Error
+myVar = 2; // OK
+
 // Functions are first class citizens, support the lambda "fat arrow" syntax and
```
Comment (kevinnls) [inline typescript.html.markdown:62]: would help to add that they're called "arrow functions" in the community

**Comment by mxschumacher** [comment]
I'm not convinced that "name1" is an improvement - if we were to treat it as production code, it would be an example of bad naming. 

The TS page isn't supposed to be run as one big block - it's just for illustration of concepts, you wouldn't copy the entire page into your code, so I don't believe the colliding names in the scope are an issue.

**Comment by kevinnls** [comment]
righto. i'll leave that to the collaborators then. shall i make a separate pr for the other changes that i suggested?

**Comment by mxschumacher** [comment]
@kevinnls you mean the arrow function comment & capitalizing Person? I think these are very small improvements, maybe your time is better spent on more impactful issues. Plenty of files in this repo would benefit from better examples & clearer explanations.

@vitiral Given my comment above, I'd like to close this PR. Is that okay for you?

**Pull request closed without merging**